yarp/yarp_compiler.c

```diff
@@ -1240,6 +1240,38 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
 
           return;
       }
+      case YP_NODE_SOURCE_ENCODING_NODE: {
+          const char *encoding = compile_context->parser->encoding.name;
+          if (!popped) {
+              // TODO: This encoding isn't an exact match of what CRuby outputs, for
+              // example "utf8" vs "#<Encoding:UTF-8>". Look into if there's a standard
+              // conversion to the expected format here.
+              ADD_INSN1(ret, &dummy_line_node, putobject, rb_str_new_cstr(encoding));
```

Comment:
I think you want `rb_find_encoding` here, but you need to convert the `encoding` into a string first.